libfreerdp/core/info.c

```diff
@@ -772,6 +762,9 @@ static void rdp_write_info_packet(rdpRdp* rdp, wStream* s)
 		cbDomain = 0;
 	}
 
+	/* excludes (!) the length of the mandatory null terminator */
+	cbDomain = cbDomain > 0 ? cbDomain - 2 : 0;
```

Comment:
I´d change that to `cbDomain = cbDomain > 2 ? cbDomain - 2 : cbDomain;` to avoid a possible overflow

Reply:
If settings->Domain is an empty string, the result cbDomain would be 2. We still need to eliminate the null terminator.

What about change that to `cbDomain = cbDomain >= 2 ? cbDomain - 2 : cbDomain;`